### DISCUSSION TITLE
How can I pass an argument with semicolons into vcpkg_build_msbuild?

### DISCUSSION THREAD

**Discussion by tjrileywisc** [microsoft/vcpkg#24818] [Q&A]
I'm trying to make a portfile with a library that requires me to pass multiple include paths into the msbuild command in order to get the build to work. I have something like the below:

```
set(ADDITIONAL_INCLUDES "${CURRENT_INSTALLED_DIR}/include/glib-2.0;${CURRENT_INSTALLED_DIR}/lib/glib-2.0/include")

vcpkg_build_msbuild(
    PROJECT_PATH ${SOURCE_PATH}
    USE_VCPKG_INTEGRATION
    OPTIONS
        /p:IncludePath="${ADDITIONAL_INCLUDES}"
)
```

The resulting msbuild command fails, since the variable with the ; is getting split by CMake into multiple args, and msbuild thinks I'm trying to build two projects. I've tried multiple permutations of backslashes to escape that semicolon to no avail. Is there a way to do this?

**Comment by dg0yt** [comment]
`;` is the cmake list separator. Did you try to escape it? I guess it takes two `\`.

In addition, be carefull with lines like this:
> ~~~
>         /p:IncludePath="${ADDITIONAL_INCLUDES}"
> ~~~
These inner quotes do not mark a string expression for CMake. This may lead to surprising results at least in cross-platform contexts, in particular when the variable contains chars like space. Most of the times, you should use quotes around the whole CMake expression:
~~~
        "/p:IncludePath=${ADDITIONAL_INCLUDES}"
~~~
If you still need quotes in the expressed value, they need to be escaped:
~~~
        "/p:IncludePath=\"${ADDITIONAL_INCLUDES}\""
~~~

**Reply by tjrileywisc** [reply]
I made the following changes:

```
set(ADDITIONAL_INCLUDES "${CURRENT_INSTALLED_DIR}/include/glib-2.0\\;${CURRENT_INSTALLED_DIR}/lib/glib-2.0/include")

vcpkg_build_msbuild(
    PROJECT_PATH ${SOURCE_PATH}
    USE_VCPKG_INTEGRATION
    OPTIONS
        "/p:IncludePath=\"${ADDITIONAL_INCLUDES}\""
)
```

Now msbuild thinks that parameter is invalid, because it's separated by `\;` instead of just `;`

```
/p:IncludePath="C:/src/vcpkg/installed/x64-windows/include/glib-2.0\;C:/src/vcpkg/installed/x64-windows/lib/glib-2.0/include"
```

I even tried `\\\;` (which I saw in some SO answers) and the parameter is separated by `\\;` now.
